ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetUncleCountByBlockHash.java

```diff
@@ -42,7 +43,7 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     final Hash hash;
     try {
       hash = requestContext.getRequiredParameter(0, Hash.class);
-    } catch (Exception e) { // TODO:replace with JsonRpcParameter.JsonRpcParameterException
+    } catch (JsonRpcParameter.JsonRpcParameterException e) {
```

Comment:
would it be nicer to modify getRequiredParameter() to throw the ex 

hash = requestContext.getRequiredParameterOrElseThrow(0, Hash.class,  "Invalid block hash parameter (index 0)", RpcErrorType.INVALID_BLOCK_HASH_PARAMS)

but there's a couple of spots where we do something different

Reply:
That would be another way to get the details we want into the exception, but in general, I'd prefer to keep low level code as simple as possible, and let higher level code handle the specifics in the appropriate context.

As a middle ground, I could implement a set of methods in JsonRpcRequestContext like 
`public Hash getRequiredHashParameter(int index)` which would clean up the higher level code, and still add detail to exceptions in the appropriate context.

Reply:
it would simplify slightly - I have no strong opinion on that - will leave up to you

Reply:
I think given the length of time it would take, it's probably not worth doing right now

Reply:
nice suggestion, it would make the code more readable removing a lot of try/catch blocks. but agree that it's probably not worth doing right now